patches/poco.patch

```diff
@@ -186,15 +186,15 @@
 +#
 +# Define Tools
 +#
-+CC      = $(TOOL)-gcc
-+CXX     = $(TOOL)-g++
++CC      = $(TOOL)-clang
++CXX     = $(TOOL)-clang++
```

Comment:
My system does not have `aarch64-linux-android-clang` and `aarch64-linux-android-clang++`. The compilers bundled in the Android NDK are located at
```
CC      = $(ANDROID_NDK)/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
CXX     = $(ANDROID_NDK)/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++
```

Reply:
`build_library_with_toolchain` copies a `toolchain` folder to the library being compiled. Then, the `<lib>/toolchain/bin` is added to the PATH:
https://github.com/Intermodalics/ros_android/blob/bb198a12759544b749cb7c7e58d8de84c9bddfe5/build_library_with_toolchain.sh#L27-L30

When I run the script individually for `poco`, this is what I see:
```
juan@Sullust:~/projects/ros_android/result_ws/libs/poco-1.8.0$ ls toolchain/bin/ -l
total 259412
-rwxrwxr-x 1 juan juan      171 nov  2 17:37 2to3
-rwxrwxr-x 1 juan juan  1244192 nov  2 17:37 aarch64-linux-android-addr2line
-rwxrwxr-x 1 juan juan  1276872 nov  2 17:37 aarch64-linux-android-ar
-rwxrwxr-x 1 juan juan  1837360 nov  2 17:37 aarch64-linux-android-as
-rwxrwxr-x 1 juan juan  1239904 nov  2 17:37 aarch64-linux-android-c++filt
-rwxrwxr-x 1 juan juan      220 nov 12 15:10 aarch64-linux-android-clang
-rwxrwxr-x 1 juan juan      224 nov 12 15:10 aarch64-linux-android-clang++
-rwxrwxr-x 1 juan juan  4630096 nov  2 17:37 aarch64-linux-android-dwp
-rwxrwxr-x 1 juan juan    32072 nov  2 17:37 aarch64-linux-android-elfedit
-rwxrwxr-x 1 juan juan      224 nov 12 15:10 aarch64-linux-android-g++
-rwxrwxr-x 1 juan juan      220 nov 12 15:10 aarch64-linux-android-gcc
-rwxrwxr-x 1 juan juan  1321216 nov  2 17:37 aarch64-linux-android-gprof
-rwxrwxr-x 1 juan juan  2465136 nov  2 17:37 aarch64-linux-android-ld
-rwxrwxr-x 1 juan juan  2465136 nov  2 17:37 aarch64-linux-android-ld.bfd
-rwxrwxr-x 1 juan juan  7972608 nov  2 17:37 aarch64-linux-android-ld.gold
-rwxrwxr-x 1 juan juan  1257184 nov  2 17:37 aarch64-linux-android-nm
-rwxrwxr-x 1 juan juan  1481888 nov  2 17:37 aarch64-linux-android-objcopy
-rwxrwxr-x 1 juan juan  2077344 nov  2 17:37 aarch64-linux-android-objdump
-rwxrwxr-x 1 juan juan  1276904 nov  2 17:37 aarch64-linux-android-ranlib
-rwxrwxr-x 1 juan juan   609944 nov  2 17:37 aarch64-linux-android-readelf
-rwxrwxr-x 1 juan juan  1244096 nov  2 17:37 aarch64-linux-android-size
-rwxrwxr-x 1 juan juan  1244192 nov  2 17:37 aarch64-linux-android-strings
-rwxrwxr-x 1 juan juan  1481920 nov  2 17:37 aarch64-linux-android-strip
-rwxrwxr-x 1 juan juan    10450 nov  2 17:37 bisect_driver.py
-rwxrwxr-x 1 juan juan      220 nov 12 15:10 clang
-rwxrwxr-x 1 juan juan      224 nov 12 15:10 clang++
-rwxrwxr-x 1 juan juan 53695608 nov  2 17:37 clang70
-rwxrwxr-x 1 juan juan 53695608 nov  2 17:37 clang70++
-rwxrwxr-x 1 juan juan  1650448 nov  2 17:37 clang-format
-rwxrwxr-x 1 juan juan     6402 nov  2 17:37 clang-tidy
-rwxrwxr-x 1 juan juan 25469704 nov  2 17:37 clang-tidy.real
-rwxrwxr-x 1 juan juan     2945 nov  2 17:37 gcore
-rwxrwxr-x 1 juan juan       94 nov  2 17:37 gdb
-rwxrwxr-x 1 juan juan  8584208 nov  2 17:37 gdb-orig
-rwxrwxr-x 1 juan juan    21811 nov  2 17:37 git-clang-format
-rwxrwxr-x 1 juan juan      169 nov  2 17:37 idle
-rwxrwxr-x 1 juan juan 29165616 nov  2 17:37 ld.lld
-rwxrwxr-x 1 juan juan  7704784 nov  2 17:37 llvm-ar
-rwxrwxr-x 1 juan juan  2146376 nov  2 17:37 llvm-as
-rwxrwxr-x 1 juan juan  2568592 nov  2 17:37 llvm-cov
-rwxrwxr-x 1 juan juan  1572688 nov  2 17:37 llvm-dis
-rwxrwxr-x 1 juan juan  2572408 nov  2 17:37 llvm-link
-rwxrwxr-x 1 juan juan  2043632 nov  2 17:37 llvm-modextract
-rwxrwxr-x 1 juan juan  7843976 nov  2 17:37 llvm-nm
-rwxrwxr-x 1 juan juan  2190728 nov  2 17:37 llvm-objcopy
-rwxrwxr-x 1 juan juan  1003488 nov  2 17:37 llvm-profdata
-rwxrwxr-x 1 juan juan  3227864 nov  2 17:37 llvm-readobj
-rwxrwxr-x 1 juan juan  2190728 nov  2 17:37 llvm-strip
-rwxrwxr-x 1 juan juan  2650264 nov  2 17:37 llvm-symbolizer
-rwxrwxr-x 1 juan juan   162536 nov  2 17:37 make
-rwxrwxr-x 1 juan juan    38344 nov  2 17:37 ndk-depends
-rwxrwxr-x 1 juan juan       87 nov  2 17:37 ndk-gdb
-rwxrwxr-x 1 juan juan    27110 nov  2 17:37 ndk-gdb.py
-rwxrwxr-x 1 juan juan   979992 nov  2 17:37 ndk-stack
-rwxrwxr-x 1 juan juan     2167 nov  2 17:37 ndk-which
-rwxrwxr-x 1 juan juan      154 nov  2 17:37 pydoc
-rwxrwxr-x 1 juan juan  2172904 nov  2 17:37 python
-rwxrwxr-x 1 juan juan  2172904 nov  2 17:37 python2
-rwxrwxr-x 1 juan juan  2172904 nov  2 17:37 python2.7
-rwxrwxr-x 1 juan juan     1744 nov  2 17:37 python2.7-config
-rwxrwxr-x 1 juan juan     1744 nov  2 17:37 python2-config
-rwxrwxr-x 1 juan juan     1744 nov  2 17:37 python-config
-rwxrwxr-x 1 juan juan     3135 nov  2 17:37 python-config.sh
-rwxrwxr-x 1 juan juan  8642784 nov  2 17:37 sancov
-rwxrwxr-x 1 juan juan  2612976 nov  2 17:37 sanstats
-rwxrwxr-x 1 juan juan    53865 nov  2 17:37 scan-build
-rwxrwxr-x 1 juan juan     4504 nov  2 17:37 scan-view
-rwxrwxr-x 1 juan juan    18617 nov  2 17:37 smtpd.py
-rwxrwxr-x 1 juan juan   666288 nov  2 17:37 yasm
```

That's why the script finds `aarch64-linux-android-clang`. Do you see something different in your system?

Reply:
Ah, you are right.I did not run it because Poco was already compiled for me, but the binaries had the host architecture. Your patch works for me after having removed the libraries and Poco sources from my target folder.